metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -896,10 +896,9 @@ def dataset_browse_prefix(self) -> str:
         return f"/{self.config.env.lower()}{self.no_env_browse_prefix}"
 
     def _re_authenticate(self):
-        tableau_auth: Union[
-            TableauAuth, PersonalAccessTokenAuth
-        ] = self.config.get_tableau_auth(self.site_id)
-        self.server.auth.sign_in(tableau_auth)
+        # Sign-in again may not be enough because Tableau sometimes caches invalid sessions
+        # so we need to restart the Tableau Server object
```

Comment:
might be worth giving some history here e.g. "Initially we just called server.auth.sign_in(). However, it turned out that ..."

```suggestion
        # so we need to recreate the Tableau Server object
```